samples/android/qr-detection/src/org/opencv/samples/qrdetection/QrdetectionProcessor.java

```diff
@@ -0,0 +1,72 @@
+package org.opencv.samples.qrdetection;
+
+import org.opencv.core.Core;
+import org.opencv.core.CvType;
+import org.opencv.core.Mat;
+import org.opencv.core.Scalar;
+import org.opencv.core.Size;
+import org.opencv.core.Point;
+import org.opencv.imgproc.Imgproc;
+import org.opencv.objdetect.GraphicalCodeDetector;
+import org.opencv.objdetect.QRCodeDetector;
+import org.opencv.objdetect.QRCodeDetectorAruco;
+
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class is a controller for puzzle game.
+ * It converts the image from Camera into the shuffled image
+ */
+public class QrdetectionProcessor {
+    private GraphicalCodeDetector detector;
+    private static final String TAG = "QrdetectionProcessor";
+    public QrdetectionProcessor(boolean useArucoDetector) {
+        if (useArucoDetector)
+            detector = new QRCodeDetectorAruco();
+        else
+            detector = new QRCodeDetector();
+    }
+
+    /* this method to be called from the outside. it processes the frame and shuffles
+     * the tiles as specified by mIndexes array
+     */
+    public synchronized Mat qrFrame(Mat inputPicture, boolean tryDecode) {
+        List<String> decoded_info = new ArrayList<String>();
+        Mat points = new Mat();
+        boolean result;
+        if (tryDecode)
+            result = detector.detectAndDecodeMulti(inputPicture, decoded_info, points);
+        else
+            result = detector.detectMulti(inputPicture, points);
+        if (result) {
+            for (int i = 0; i < points.rows(); i++) {
```

Comment:
I propose to extract rendering as method and introduce Scalars for colors as named constatnts to not re-create them each time.